snet_cli/mpe_service_metadata.py

```diff
@@ -171,6 +171,14 @@ def get_payment_address(self, group_name = None):
     def get_all_endpoints(self):
         return [e["endpoint"] for e in self.m["endpoints"]]
 
+    def get_all_endpoints_with_group_name(self):
+        endpts_with_grp = {}
```

Comment:
You could use defaultdict (defaultdict(list)) to simplify code:
```python
from collections import defaultdict 
....
def get_all_endpoints_with_group_name(self):
        endpts_with_grp = defaultdict(list)
        for e in self.m["endpoints"]:
            endpts_with_grp[e['group_name']].append(e['endpoint'])
        return endpts_with_grp
```

Reply:
noted and updated. Will make use of defaultdict, it makes the code simpler.